astropy/utils/data_info.py

```diff
@@ -768,19 +779,19 @@ def get_sortable_arrays(self):
 
 class MixinInfo(BaseColumnInfo):
     @property
-    def name(self):
+    def name(self) -> str | None:
         return self._attrs.get("name")
 
     @name.setter
-    def name(self, name):
+    def name(self, name: str | None):
         # For mixin columns that live within a table, rename the column in the
         # table when setting the name attribute.  This mirrors the same
         # functionality in the BaseColumn class.
+        new_name = None if name is None else str(name)
```

Comment:
It seems like this can exactly follow the pattern in the `Column.name` setter, i.e.
```
if name is not None:
    name = str(None)
```
I'm not sure why the original is different.